native/core/src/parquet/parquet_support.rs

```diff
@@ -1861,6 +1863,42 @@ fn trim_end(s: &str) -> &str {
     }
 }
 
+// Default object store which is local filesystem
+#[cfg(not(feature = "hdfs"))]
+pub(crate) fn register_object_store(
+    session_context: Arc<SessionContext>,
+) -> Result<ObjectStoreUrl, ExecutionError> {
+    let object_store = object_store::local::LocalFileSystem::new();
+    let url = ObjectStoreUrl::parse("file://")?;
+    session_context
+        .runtime_env()
+        .register_object_store(url.as_ref(), Arc::new(object_store));
+    Ok(url)
+}
+
+// HDFS object store
+#[cfg(feature = "hdfs")]
+pub(crate) fn register_object_store(
+    session_context: Arc<SessionContext>,
+) -> Result<ObjectStoreUrl, ExecutionError> {
+    // TODO: read the namenode configuration from file schema or from spark.defaultFS
```

Comment:
Do we need to register object store from `native_scan.file_partitions`?

Reply:
Thanks @wForget I'm not sure I'm getting it, do you mean the better place to register the object store will be inside `file_partitions` iterator loop ?

Reply:
> do you mean the better place to register the object store will be inside file_partitions iterator loop ?

Yes, is it possible that native scan paths correspond to multiple object stores or are different from spark.defaultFs?

Reply:
for HDFS/S3 the default fs can be taken from `spark.hadoop.fs.defaultFS` parameter. 
To support multiple object stores that is interesting idea however I'm not sure when it can be addressed

Reply:
> for HDFS/S3 the default fs can be taken from spark.hadoop.fs.defaultFS parameter. 

Sometimes I also access other hdfs ns like:

```
select * from `parquet`.`hdfs://other-ns:8020/warehouse/db/table`
```

Reply:
that is interesting scenario, I'll add a separate test case for this